src/devices/telldus.c

```diff
@@ -0,0 +1,182 @@
+	uint8_t channel;
+	uint8_t humidity;
+	uint8_t message[PACKETLEN-1];
+	int tmp;
+	float temperature;
+	data_t *data;
+
+	/* Correct number of rows? */
+	if (bitbuffer->num_rows != 2) {
+		if (debug_output) {
+			fprintf(stderr, "%s %s: wrong number of rows (%d)\n", 
+				time_str, __func__, bitbuffer->num_rows);
+		}
+		return 0;
+	}
+
+	/* Correct bit length? */
+	if (bitbuffer->bits_per_row[1] != BITLEN) {
+		if (debug_output) {
+			fprintf(stderr, "%s %s: wrong number of bits (%d)\n", 
+				time_str, __func__, bitbuffer->bits_per_row[0]);
+		}
+		return 0;
+	}
+
+	bb = bitbuffer->bb[1];
+
+	/* Correct start sequence? */
+	if (bb[0] != STARTBYTE) {
+		if (debug_output) {
+			fprintf(stderr, "%s %s: wrong start byte\n", time_str, __func__);
+		}
+		return 0;
+	}
+
+	/* If debug enabled, print the raw information */
+	if (debug_output) {
+		fprintf(stderr, "%s %s: recieved data = ", time_str, __func__);
+		for (i = 0; i < PACKETLEN; i++) {
+			fprintf(stderr, "%02x ",bb[i]);
+		}
+		fprintf(stderr, "\n");
+	}
+
+	/* Read message for crc calculation */
+	for (i = 0; i < PACKETLEN-1; i++) {
+		message[i] = bb[i];
+	}
+
+	/* Correct CRC? */
+	r_crc = bb[PACKETLEN - 1]; /* Last byte in the data */
+	c_crc = crc8(message, PACKETLEN-1, 0x31, INIT_CRC);
+
+	if (r_crc != c_crc) {
+		if (debug_output) {
+			fprintf(stderr, "%s %s: CRC failed, calculated %x, received %x\n",
+				time_str, __func__, c_crc, r_crc);
+		}
+		return 0;
+	}
+
+	/* Message validated, now parse the data */
+
+	/* Invert bitbuffer */
+	bitbuffer_invert(bitbuffer);
+	bb = bitbuffer->bb[1];
+
+	/* Sensor id */
+	sensor_id = (bb[2] >> 4) & 0x0f;
+	/* Let's rename sensor 0111 to 1, and 1000 to 2 */
+	if(sensor_id == 0x07)
+		sensor_id = 0x01;
+	else
+		sensor_id = 0x02;
+
+	/* Channel */
+	channel = (bb[1] & 0x07) + 1;
+
+	/* Temperature */
+	tmp = bb[PACKETLEN-4] & 0x0f;
+	tmp <<= 8;
+	tmp |= bb[PACKETLEN-3];
+	if(tmp & 0x800) { /* if sign bit is 1 */
+		tmp &= 0x7ff;
+		tmp = ~tmp + 1;
+	}
+	temperature = tmp / 10.0f;
+
+	/* Humidity */
+	humidity = bb[PACKETLEN-2];
+
+	/* Get the time */
+	local_time_str(0, time_str);
+
+	if (humidity != 0xff) { /* if there is humidity sensor */
+		data = data_make("time",			"",				DATA_STRING,	time_str,
+						 "model",			"",				DATA_STRING,	"Telldus/Proove thermometer",
+						 "channel",			"Channel",		DATA_INT,		channel,
+						 "sensor", 			"Sensor id",	DATA_INT,		sensor_id,
+						 "temperature_C",	"Temperature",	DATA_FORMAT,	"%.1f C", DATA_DOUBLE, temperature,
+						 "humidity",		"Humidity",		DATA_FORMAT,	"%i%%", DATA_INT, humidity,
+						 NULL);
+	} else {
+		data = data_make("time",			"",				DATA_STRING,	time_str,
+						 "model",			"",				DATA_STRING,	"Telldus/Proove thermometer",
+						 "channel",			"Channel",		DATA_INT,		channel,
+						 "sensor", 			"Sensor id",	DATA_INT,		sensor_id,
+						 "temperature_C",	"Temperature",	DATA_FORMAT,	"%.1f C", DATA_DOUBLE, temperature,
+						 "humidity",		"Humidity",		DATA_STRING,	"N/A",
```

Comment:
best to remove this line "N/A" is not what parser for humidity data would expect. No sensor value: no output :)